lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1418 @@
+      let response1 = null;
+
+      // Timeout might be zero, but the callback will fire anyway
+      // We use sendMessage instead of super.send() because we want to pull raw
+      // data only, and we can't determine response types until after we've checked
+      // them. The CNL can send the messages in different orders.
+      this.sendMessage(hidDevice)
+        .then(() => BCNLMessage.wait(sendDelay))
+        .then(() => this.readMessage(hidDevice, readTimeout))
+        .then((response) => {
+          response1 = response;
+        })
+        .then(() => BCNLMessage.wait(sendDelay))
+        .then(() => this.readMessage(hidDevice, readTimeout))
+        .then((response2) => {
+          let astmInfo = '';
+
+          if (response1[0] === BCNLMessage.ASCII_CONTROL.EOT) {
+            astmInfo = Buffer.from(response1).toString('ascii');
+            new BCNLCommandResponse(response2).checkAsciiControl(BCNLMessage.ASCII_CONTROL.ENQ);
+          } else {
+            astmInfo = Buffer.from(response2).toString('ascii');
+            new BCNLCommandResponse(response1).checkAsciiControl(BCNLMessage.ASCII_CONTROL.ENQ);
+          }
+          const message = new DeviceInfoRequestResponse(astmInfo);
+          resolve(message);
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+  }
+}
+
+class MinimedPumpSession {
+  constructor(bcnlModelAndSerial) {
+    this.envelopeSequenceNumber = 0;
+    this.telDSequenceNumber = 0;
+    this.comDSequenceNumber = 0;
+    this.bcnlModelAndSerial = bcnlModelAndSerial;
+    this.radioChannel = null;
+    this.linkMAC = null;
+    this.pumpMAC = null;
+    this.key = null;
+    this.pumpModel = null;
+    this.pumpSerial = null;
+  }
+
+  get bcnlSerialNumber() {
+    return this.bcnlModelAndSerial.replace(/\d+-/, '');
+  }
+
+  get iv() {
+    const iv = Buffer.from(this.key);
+    iv[0] = this.radioChannel;
+
+    return iv;
+  }
+
+  get packedLinkMAC() {
+    return Buffer.from(this.linkMAC, 'hex').swap64().toString('binary');
+  }
+
+  get packedPumpMAC() {
+    return Buffer.from(this.pumpMAC, 'hex').swap64().toString('binary');
+  }
+
+  getHMAC() {
+    const paddingKey = 'A4BD6CED9A42602564F413123';
+    const digest = crypto.createHash('sha256')
+      .update(`${this.bcnlSerialNumber}${paddingKey}`)
+      .digest()
+      .reverse();
+    return digest;
+  }
+}
+
+class MinimedMessage extends BCNLMessage {
+  static get COMMAND_TYPE() {
+    return {
+      OPEN_CONNECTION: 0x10,
+      CLOSE_CONNECTION: 0x11,
+      SEND_MESSAGE: 0x12,
+      READ_INFO: 0x14,
+      REQUEST_LINK_KEY: 0x16,
+      SEND_LINK_KEY: 0x17,
+      RECEIVE_MESSAGE: 0x80,
+      SEND_MESSAGE_RESPONSE: 0x81,
+      REQUEST_LINK_KEY_RESPONSE: 0x86,
+    };
+  }
+
+  static get ENVELOPE_SIZE() {
+    return 33;
+  }
+
+  static get MINIMED_HEADER() {
+    return 0x5103; // Q\x03
+  }
+
+  static oneByteChecksum(buffer) {
+    // eslint-disable-next-line no-bitwise
```

Comment:
For this driver, I'm using the `.eslintrc` from `viz`, which is much stricter. My thought is that explicitly calling out `no-bitwise` is a good rule of thumb, even in files where we do a lot of bitwise operations, because it helps reduce any logic bugs.